### PR TITLE
Add ability to set the boot order on vm

### PR DESCRIPTION
**Pull request by ChrisPortman**
Add boot_order=(devices) method to vm class.  Accepts an array of boot options e.g:
  [ :network, :cdrom, :hd ]

### CODE REVIEW

**Comment by Fryguy** [comment]
@brandondunne Please review.

File: lib/ovirt/vm.rb [hunk 1]
```diff
@@ -80,6 +80,18 @@ def move(storage_domain)
...
 
+    def boot_order=(devices)
+      devices = [ devices ].flatten
+      update! do |xml|
+        xml.os do
+          devices.each do |d|
+            xml.boot(:dev => d)
+          end
+        end
+        xml.memory self[:memory]  # HACK: RHEVM BUG: RHEVM resets it to 10GB, unless we set it
```
Comment (brandondunne) [inline lib/ovirt/vm.rb]: Is this necessary?  I didn't see any problems in my testing against v3.3 or v3.4.  I don't remember if this problem was in updating specific attributes or if it was version dependent.

**Comment by brandondunne** [comment]
@ChrisPortman Great change!  Can you add a test for it too?

File: lib/ovirt/vm.rb (hunk 1 above)
Comment (ChrisPortman) [inline lib/ovirt/vm.rb]: I'm really not sure about this, I saw that all the similar methods had this so I added it for consistency.

Comment (brandondunne) [inline lib/ovirt/vm.rb]: Let's remove it.  If we need it, we can add it later (with a better description of the problem).

**Comment by ChrisPortman** [comment]
Hi guys,

I've added tests, removed the "hack", and fixed some style errors that the CI checks were complaining about.

Thanks,
Chris

**Comment by brandondunne** [comment]
Thanks @ChrisPortman :+1:

**Pull request merged**